### PR TITLE
MTSJ with minimal fixes to support oasp4j 3.0 

### PR DESCRIPTION
**Pull request by vapadwal**

### CODE REVIEW

File: java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/impl/dao/BookingDaoImpl.java [hunk 1]
```diff
@@ -43,7 +44,7 @@ public BookingDaoImpl() {
 
     BookingEntity booking = Alias.alias(BookingEntity.class);
     EntityPathBase<BookingEntity> alias = Alias.$(booking);
-    JPAQuery query = new JPAQuery(getEntityManager()).from(alias);
+    JPAQuery query = (JPAQuery) new JPAQuery(getEntityManager()).from(alias);
```
Comment (hohwille) [inline java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/impl/dao/BookingDaoImpl.java]: This cast looks pointless to me. Also JPAQuery is now generified. Should IMHO be `JPAQuery<BookingEntity>`

File: java/mtsj/core/src/main/resources/db/specific/h2/V0001__R001_Create_schema.sql [hunk 2]
```diff
@@ -197,6 +197,6 @@ CREATE TABLE BinaryObject (
 -- *** RevInfo (Commit log for envers audit trail) ***
 CREATE TABLE RevInfo(
   id BIGINT NOT NULL GENERATED BY DEFAULT AS IDENTITY (START WITH 1),
-  timestamp BIGINT NOT NULL,
+  "timestamp" BIGINT NOT NULL,
```
Comment (hohwille) [inline java/mtsj/core/src/main/resources/db/specific/h2/V0001__R001_Create_schema.sql]: good catch. We need to document that change in release-notes and hints for upgrade.

**Comment by hohwille** [comment]
After the update JUnit gets upgraded to version 5. In another project that I tried this resulted in this error:
https://github.com/junit-team/junit5/issues/801

**Comment by hohwille** [comment]
at least I get this:
```
[WARNING] Corrupted STDOUT by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file ...\target\surefire-reports\2018-08-14T19-10-01_074-jvmRun1.dumpstream
...
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project ...: There are test failures.
...
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
```

**Comment by hohwille** [comment]
OK, Problem resolved. Root cause was too little RAM left and OOME.

**Comment by vapadwal** [comment]
Before merging it please merge https://github.com/oasp/oasp4j/pull/673 .

Also kept <oasp4j.version>3.0.0-SNAPSHOT</oasp4j.version> this in pom.xml, will be removed when we have oasp4j 3.0 release

**Comment by vapadwal** [comment]
Updated it with latest changes according to PR https://github.com/oasp/oasp4j/pull/675

**Comment by vapadwal** [comment]
Closing this PR as there is a new PR which contains all the migration https://github.com/oasp/my-thai-star/pull/135

**Pull request closed without merging**